### DISCUSSION TITLE
SD friendly F2FS for system image?

### DISCUSSION THREAD

**Discussion by spitfire** [jayofelony/pwnagotchi#10] [Ideas]
I’ve converted my installation to use f2fs instead of ext4 to lessen the wear on the SD card. I think it could be a good idea for the image to use it as well - that would probably make using workarounds such as putting stuff in zram unnecessary.

**Comment by kagehisa** [comment]
Could you create a PR for this? I would be interested as well and since you seem to have it running a PR would be nice.

**Comment by spitfire** [comment]
I know how to do it on an existing system, but don't know enough about the build system that generates the images to submit a PR.
You can follow [this guide](http://whitehorseplanet.org/gate/topics/documentation/public/howto_ext4_to_f2fs_root_partition_raspi.html) which I have been using for a long time on Raspberry Pi OS. If you have an SD card with lots of free space you can add overprovision parameter (i.e. `sudo mkfs.f2fs -o 50 /dev/sdb2`) to be safer in case some sectors die on your card. You can further lessen wear on the card by [doing this](https://github.com/jayofelony/pwnagotchi-bookworm/discussions/11).

**Comment by jayofelony** [comment]
I'll see if I can build an image using f2fs. Swap could be usefull for auto-updates though, can be quite memory exhausting.

**Reply by jayofelony** [reply]
Won´t work sadly, as the base image is ext4. So this has to be done manually if you want to.
